lib/semantics/check-do.cc

```diff
@@ -139,18 +140,13 @@ class DoConcurrentBodyEnforce {
   void Post(const parser::UnlockStmt &) { NoImageControl(); }
   void Post(const parser::StopStmt &) { NoImageControl(); }
 
-  void Post(const parser::AllocateStmt &) {
-    if (anyObjectIsCoarray()) {
-      context_.Say(currentStatementSourcePosition_,
-          "ALLOCATE coarray not allowed in DO CONCURRENT"_err_en_US);
-    }
+  void Post(const parser::AllocateStmt &allocateStmt) {
+    CheckDoesntContainCoarray(allocateStmt);
```

Comment:
Maybe you could actually add a reference to C1137 here and in `Post(const parser::DeallocateStmt &)`. It is rather clear, but I guess it does not hurt adding it. Sorry for not commenting on this during the first review.

Reply:
Good idea.  I'll add the references.